### ISSUE TITLE
Am I mis-understanding the meaning of MaskIoU head here?

### ISSUE THREAD

**Issue by CrazySherman** [zjhuang22/maskscoring_rcnn#10]
I checked the paper and really thx for the awesome work and code share!
Previously I thought "MaskIoU" is that you're training the mask with IoU directly (dice loss) -- your predict a mask for box i of class A, and you use the ground of box i of class A to compute IoU and use that as a loss to train your mask-rcnn in a multi-task learning fashion.
Checking the paper more carefully I realized you're feeding the predicted mask to a CNN to compute IoU rather than compute it directly as https://github.com/kevinzakka/pytorch-goodies/blob/master/losses.py#L54  I wonder if this approach is prone to overfitting, have you compared the results with/without "concatenation" and see how much the MaskIoU head is actually relying on RoI feature maps?

**Comment by zjhuang22** [comment]
> I checked the paper and really thx for the awesome work and code share!
> Previously I thought "MaskIoU" is that you're training the mask with IoU directly (dice loss) -- your predict a mask for box i of class A, and you use the ground of box i of class A to compute IoU and use that as a loss to train your mask-rcnn in a multi-task learning fashion.
> Checking the paper more carefully I realized you're feeding the predicted mask to a CNN to compute IoU rather than compute it directly as https://github.com/kevinzakka/pytorch-goodies/blob/master/losses.py#L54 I wonder if this approach is prone to overfitting, have you compared the results with/without "concatenation" and see how much the MaskIoU head is actually relying on RoI feature maps?

Thanks for your attention. In the paper, we present the results of ResNet-18: mAP=29.3. We have tried only use the mask (without RoI feature) to predict IoU: mAP=28.2. That shows the RoI features are important.
